rules/Transform/ValueObject/PropertyFetchToMethodCall.php

```diff
@@ -47,4 +52,30 @@ public function getNewGetArguments(): array
     {
         return $this->newGetArguments;
     }
+
+    public function isEquivalent(): bool
+    {
+        if($this->oldProperty !== self::EQUIVALENT) {
+            return false;
+        }
+
+        if($this->newGetMethod !== self::EQUIVALENT) {
+            return false;
+        }
+
+        return  $this->newSetMethod === self::EQUIVALENT;
+    }
```

Comment:
I think new users will never notice this special behavior and would fail to use rule as you design.
Is there a way to handle this without extra methods in value object?


<br>

The developer experience of using this rule should be as smooth as:

```php
new PropertyFetchToMethodCall('SomeObject', 'property'),
```

How can we make it this easy?

Reply:
I totally agree. But i tried to add new behaviour to the existing one. I didn´t understand the old rule design either. Would love to redesign the rule. But you can maybe say more about the original use case of the old rule.

Reply:
I have never used it and i was thinking about splitting it up. But you said in the issue https://github.com/rectorphp/rector/issues/6440 this rule should handle the new behaviour. I am little bit lost now.

Reply:
I understand the original behaviour. A property fetch can always be used on both sides. The reading and the writing one. But maybe we should split those.

Reply:
Feel free to change design of rule to make it accept simple configuration 👍